src/components/data/DataSegment/DataSegment.tsx

```diff
@@ -26,40 +73,51 @@ import './DataSegment.css';
  * - `DataSegment.Item` = segment item
  * - `DataSegment.Action` = segment action area
  */
-export default class DataSegment extends Component {
-    // @ts-expect-error TS(7006) FIXME: Parameter 'props' implicitly has an 'any' type.
-    constructor(props, context) {
+export default class DataSegment extends Component<DataSegmentProps, DataSegmentState> {
+    static Item = SegmentItem;
+
+    static Action = DataAction;
+
+    // eslint-disable-next-line react/static-property-placement
+    static defaultProps = {
```

Comment:
Can we have here:
```suggestion
    static defaultProps: Partial<DataSegmentProps> = {
```
? It should make it more secure.

Reply:
sadly, no, because then the typing system doesnt correctly do the magic to make things optional for the whole component

Reply:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#explicit-types-on-defaultprops

Reply:
Got it, thanks for the link. Didn't know that TypeScript recognizes `defaultProps` this way.